extensions/interactions/FractionInput/directives/oppia-interactive-fraction-input.component.ts

```diff
@@ -0,0 +1,191 @@
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Directive for the FractionInput interaction.
+ */
+
+import { Component, Input, OnDestroy, OnInit } from '@angular/core';
+import { Subject, Subscription } from 'rxjs';
+import { debounceTime, distinctUntilChanged } from 'rxjs/operators';
+
+import { FractionObjectFactory } from 'domain/objects/FractionObjectFactory';
+import { ObjectsDomainConstants } from 'domain/objects/objects-domain.constants';
+import { FractionInputCustomizationArgs } from 'interactions/customization-args-defs';
+import { InteractionAttributesExtractorService } from 'interactions/interaction-attributes-extractor.service';
+import { InteractionRulesService } from 'pages/exploration-player-page/services/answer-classification.service';
+import { CurrentInteractionService } from 'pages/exploration-player-page/services/current-interaction.service';
+
+import { FractionInputRulesService } from './fraction-input-rules.service';
+import { downgradeComponent } from '@angular/upgrade/static';
+
+@Component({
+  selector: 'oppia-interactive-fraction-input',
+  templateUrl: './fraction-input-interaction.component.html',
+  styleUrls: []
+})
+export class InteractiveFractionInputComponent implements OnInit, OnDestroy {
+  @Input() requireSimplestFormWithValue: string = '';
+  @Input() allowImproperFractionWithValue: string = '';
+  @Input() allowNonzeroIntegerPartWithValue: string = '';
+  @Input() customPlaceholderWithValue: string = '';
+  componentSubscriptions: Subscription = new Subscription();
+  requireSimplestForm: boolean = false;
+  allowImproperFraction: boolean = true;
+  allowNonzeroIntegerPart: boolean = true;
+  customPlaceholder: string = '';
+  FORM_ERROR_TYPE: string = 'FRACTION_FORMAT_ERROR';
+  errorMessage: string = '';
+  answer: string = '';
+  isValid: boolean = true;
+  answerChanged: Subject<string> = new Subject<string>();
+  FRACTION_INPUT_FORM_SCHEMA = {
+    type: 'unicode',
+    ui_config: {}
+  };
+  constructor(
+    private currentInteractionService: CurrentInteractionService,
+    private fractionInputRulesService: FractionInputRulesService,
+    private fractionObjectFactory: FractionObjectFactory,
+    private interactionAttributesExtractorService:
+      InteractionAttributesExtractorService
+  ) {
+    this.componentSubscriptions.add(this.answerChanged.pipe(
+      // Wait 150ms after the last event before emitting last event.
+      debounceTime(150),
+      // Only emit if value is different from previous value.
+      distinctUntilChanged()
+    ).subscribe(newValue => {
+      const INVALID_CHARS_REGEX = /[^\d\s\/-]/g;
+      const INVALID_CHARS_LENGTH_REGEX = /\d{8,}/;
+      // Accepts incomplete fraction inputs
+      // (see examples above except last).
```

Comment:
To which examples is this referring?

Reply:
Those were in ngOnInit. I have moved them in the constructor. Now you should be able to see it.